applications/FluidDynamicsApplication/custom_elements/two_fluid_navier_stokes.cpp

```diff
@@ -164,11 +158,80 @@ void TwoFluidNavierStokes<TElementData>::CalculateLocalSystem(
                     ComputeGaussPointEnrichmentContributions(data, Vtot, Htot, Kee_tot, rhs_ee_tot);
                 }
 
+                Matrix int_shape_function, int_shape_function_enr_neg, int_shape_function_enr_pos;
+                GeometryType::ShapeFunctionsGradientsType int_shape_derivatives;
+                Vector int_gauss_pts_weights;
+                std::vector< array_1d<double,3> > int_normals_neg;
+
+                if (rCurrentProcessInfo[SURFACE_TENSION] || rCurrentProcessInfo[MOMENTUM_CORRECTION]){
+                    ComputeSplitInterface(
+                        data,
+                        int_shape_function,
+                        int_shape_function_enr_pos,
+                        int_shape_function_enr_neg,
+                        int_shape_derivatives,
+                        int_gauss_pts_weights,
+                        int_normals_neg,
+                        p_modified_sh_func);
+                }
+
+                if (rCurrentProcessInfo[MOMENTUM_CORRECTION]){
+                    MatrixType lhs_acc_correction = ZeroMatrix(NumNodes*NumNodes,NumNodes*NumNodes);
+
+                    double positive_density;
+                    double negative_density;
+
+                    GeometryType::Pointer p_geom = this->pGetGeometry();
+
+                    for (unsigned int intgp = 0; intgp < int_gauss_pts_weights.size(); ++intgp){
+                        double u_dot_n = 0.0;
+                        for (unsigned int i = 0; i < NumNodes; i++){
+                            u_dot_n += int_shape_function(intgp,i)*(*p_geom)[i].GetValue(DISTANCE_DIFFERENCE);
+                            /* for (unsigned int dim = 0; dim < NumNodes-1; dim++){
+                                u_dot_n += int_shape_function(intgp,i)*data.Velocity(i,dim)*(int_normals_neg[intgp])[dim];
+                            } */
+
+                            if (data.Distance[i] > 0.0){
+                                positive_density = data.NodalDensity[i];
+                            } else {
+                                negative_density = data.NodalDensity[i];
+                            }
+                        }
+
+                        u_dot_n /= data.DeltaTime;
+
+                        for (unsigned int i = 0; i < NumNodes; i++){
+                            for (unsigned int j = 0; j < NumNodes; j++){
+                                for (unsigned int dim = 0; dim < NumNodes-1; dim++){
```

Comment:
```suggestion
                        for (unsigned int i = 0; i < NumNodes; ++i){
                            for (unsigned int j = 0; j < NumNodes; ++j){
                                for (unsigned int dim = 0; dim < NumNodes-1; ++dim){
```
Just in case the compiler optimizer is not doing it.

Reply:
Done.